src/couch/src/couch_ejson_size.erl

```diff
@@ -32,15 +46,10 @@ ejson_size(Integer) when is_integer(Integer), Integer > 0 ->
 
 ejson_size(Integer) when is_integer(Integer), Integer < 0 ->
     % 2 is because 1 is for the - character
-    trunc(math:log10(Integer)) + 2;
+    trunc(math:log10(-Integer)) + 2;
```

Comment:
Depending how expensive log operation is, this might work well as a lookup for a first few length.

So if integer 0<=x<10 -> 1, 10<=x<100 -> 2, 100<=x<1000 -> 3, etc. for a few more sizes.